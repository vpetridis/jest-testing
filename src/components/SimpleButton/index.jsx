import React from "react"
import PropTypes from "prop-types"

const SimpleButton = ({ text }) => {
  return (
    <div>
      <button>{text}</button>
    </div>
  )
}

SimpleButton.propTypes = {
  text: PropTypes.string.isRequired,
}

export default SimpleButton
