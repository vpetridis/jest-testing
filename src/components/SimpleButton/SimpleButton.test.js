import React from "react"
import {
  render,
  screen,
} from "@testing-library/react"
import SimpleButton from "./index"

it("should render a simple button with text", function () {
  render(<SimpleButton text="Hello World" />)
  const button = screen.getByTestId("username")
  expect(button).toMatchSnapshot()
})
