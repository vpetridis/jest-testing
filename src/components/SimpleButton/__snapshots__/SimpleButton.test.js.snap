// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a simple button with text 1`] = `
<button
  data-testid="username"
  name="username"
>
  Hello World
</button>
`;
